django/http/request.py

```diff
@@ -377,10 +378,16 @@ def _load_post_and_files(self):
                 self._mark_post_parse_error()
                 raise
         elif self.content_type == "application/x-www-form-urlencoded":
-            self._post, self._files = (
-                QueryDict(self.body, encoding=self._encoding),
-                MultiValueDict(),
-            )
+            # According to RFC 1866, the "application/x-www-form-urlencoded"
+            # content type does not have a charset and should be always treated
+            # as UTF-8.
+            if self._encoding is not None and self._encoding != "utf-8":
```

Comment:
I believe we should treat the encoding as case insensitive. If a request sends UTF-8, badrequest is raised and I don't think that's what we want:
```suggestion
            if self._encoding is not None and self._encoding.lower() != "utf-8":
```
(and perhaps we should add a test for this as well)

Reply:
Fixed.